lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +38,21 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not (
+        isinstance(text, str) and
+        isinstance(start_of_word, str | None) and
```

Comment:
`None` is not a type, so `isinstance` is not a good check here, consider using `is`